MobiFlight/MobiFlightFirmwareUpdater.cs

```diff
@@ -38,31 +39,108 @@ public static bool Update(MobiFlightModule module)
             String Port = module.InitUploadAndReturnUploadPort();
             if (module.Connected) module.Disconnect();
 
-            while (!SerialPort.GetPortNames().Contains(Port))
-            {
-                System.Threading.Thread.Sleep(100);
-            }
-
             if (module.Board.AvrDudeSettings != null)
             {
-                try {
+                while (!SerialPort.GetPortNames().Contains(Port))
+                {
+                    System.Threading.Thread.Sleep(100);
+                }
+
+                try
+                {
                     RunAvrDude(Port, module.Board);
                     result = true;
-                } catch(Exception e) {
+                } catch (Exception e) {
                     result = false;
                 }
 
                 if (module.Board.Connection.DelayAfterFirmwareUpdate > 0)
                 {
                     System.Threading.Thread.Sleep(module.Board.Connection.DelayAfterFirmwareUpdate);
                 }
-            } else
+            }
+            else if (module.Board.UsbDriveSettings != null)

```

Comment:
If the USB drive settings are present then try a USB flash instead. This is basically an exact copy of the AvrDude code and could probably be refactored a bit so the test is done in the try{} block and the delay code isn't duplicated.